### PR TITLE
3.3.5 skip dk

### PR DESCRIPTION
**Pull request by danjohnso**
@acidmanifesto Noticed some quest rewards were missing and the talent points were incorrect until client was restarted.  Other issues also existed with advancing DK level beyond 58 with glyphs and stats

**Changes proposed**:

- Change quest completion list to correctly assign all talents, spells, and items from starting zone
- Cleanup handling of main skip action to single helper for the 3 possible instances
- Remove possible error from user configuration having overlapping options
- Update player more thoroughly to avoid client restarts after module runs to get updated character

**Target branch(es)**: 335

**Issues addressed**: Fixes #

**Tests performed**: (Does it build, tested in-game, etc)

Built and tested in game with all options

**Known issues and TODO list**:

None

### CODE REVIEW

File: src/server/scripts/Custom/SkipDK.cpp
Comment (acidmanifesto) [inline src/server/scripts/Custom/SkipDK.cpp]: dks are given a hearthstone when they check in with their first inn after completing the final turned of their faction's quest line dealing with the dk starter area per retail.

File: src/server/scripts/Custom/SkipDK.cpp
Comment (acidmanifesto) [inline src/server/scripts/Custom/SkipDK.cpp]: this part can work, the all save is being debated in being removed as there are some reports of deadlocks happening with high population servers of 500 or more players.

File: src/server/scripts/Custom/SkipDK.cpp
Comment (acidmanifesto) [inline src/server/scripts/Custom/SkipDK.cpp]: talent points are give when a player levels up, this would ensure accurate. 
quest items depends on if it is rewarded gears, spells are already given with the exsisting module with their appropiate quests being rewarded.
Current module ensures gear and select spells and conditions are met to ensure propper phasing of the ebon hold being phased.

File: src/server/scripts/Custom/SkipDK.cpp
Comment (acidmanifesto) [inline src/server/scripts/Custom/SkipDK.cpp:109]: This was something being debated as well of either Setlevel and GiveLevel by someone else before and i was told a Pr was going to be put incoporating it... but that was over 6 months ago and never saw the pr yet. So i didn't wanted to make that change as i did not wanted to have the impression of yanking someone's suggestion with out their appropiate gitname to credit for the suggestion.

File: src/server/scripts/Custom/SkipDK.cpp
Comment (danjohnso) [reply src/server/scripts/Custom/SkipDK.cpp]: Thought that might be the case, I didn't think it would hurt to leave this in from your implemnentation

File: src/server/scripts/Custom/SkipDK.cpp
Comment (acidmanifesto) [inline src/server/scripts/Custom/SkipDK.cpp]: is this dealing with unicode adjustments? i usually save the files as utf-8 unsigned to ensure language compatibility with the quick gossip hard code translations. I am not really a fan of diluting the db and only make the adjustments to sql as bare basic needed.

File: src/server/scripts/Custom/SkipDK.cpp
Comment (danjohnso) [reply src/server/scripts/Custom/SkipDK.cpp]: Normally talent points are awarded as you level, but DK quests specifically have talent points assigned to them.  There were not any uses of the full character load that you get from restarting the client in code that I could find, so I thought it more correct to actually complete the quests and let the counter increase "naturally" for DK talent points.
The neck piece was missing as a manually added reward.  Other items were correct.  The phasing of ebon hold looked correct when I went back to check on runeforging.  Some of the awarded spells from your implementation were from higher levels of runeforging (level 72 runes) and the Skill 960 does not exist in 3.3.5a.  Was that from retail?  Either way, I added all the quests from 3.3.5 so the manual adds of items or spells were not required.

**Comment by acidmanifesto** [comment]
@danjohnso I left a few notes pretty much explaining the choices made with me when i made the module. your choices are good, and at the time of me making the module creation, there was really nothing to base it off of, so to have another set of eyes is great. All i ask is we check any and all things to prevent regression of something missing or broken. I am not a maintainer or any role outside of just a normal user with this git, so if you ever feel like this is fully tested and checked on, let me know and I can drop a message to jackpoz in merging.
Feel free to get in direct contact with me via discord if you use. 
My discord tag is M'Dic#0865

File: src/server/scripts/Custom/SkipDK.cpp
Comment (acidmanifesto) [reply src/server/scripts/Custom/SkipDK.cpp]: also another note, just giving them the hearthstone wouldnt be wrong either. As u pretty much would of binded with a innkeeper when u get done with the area anyways.

File: src/server/scripts/Custom/SkipDK.cpp
Comment (danjohnso) [reply src/server/scripts/Custom/SkipDK.cpp:109]: Apologies to whoever had this idea originally, I think it is correct though since GiveLevel properly checks things like glyphs and stats.

File: src/server/scripts/Custom/SkipDK.cpp
Comment (danjohnso) [reply src/server/scripts/Custom/SkipDK.cpp]: Might have been visual studio on save...

File: src/server/scripts/Custom/SkipDK.cpp
Comment (acidmanifesto) [reply src/server/scripts/Custom/SkipDK.cpp:109]: I have no issues of u doing it. As it was discussed privately to me from the other person. So you making mention of it in  a git discuss ensures it was infact a original public idea on your end.

**Comment by acidmanifesto** [comment]
How would the quests dealing with reward choices work out with this adjustment?

**Comment by danjohnso** [comment]
> How would the quests dealing with reward choices work out with this adjustment?

It defaults to the first reward option.  IRC it's just choice of sword or axe and choice of resistance type on the cloak.  A further enhancement could add those choices to the lich king dialogue

File: src/server/scripts/Custom/SkipDK.cpp
Comment (danjohnso) [reply src/server/scripts/Custom/SkipDK.cpp]: New commit should fix the encoding back to UTF8

**Comment by danjohnso** [comment]
@jackpoz I think this is in a good place now unless @acidmanifesto has anything else he wants changed

**Comment by acidmanifesto** [comment]
danjohnso and myself are active discussing and exploring some changes that will pushed here soon^TM

**Comment by acidmanifesto** [comment]
@jackpoz this PR is all good and ready to go for merge

**Review by acidmanifesto — APPROVED** [review]

**Pull request merged**
